Improve alignment and spacing of various UI elements.

EreMaijala opened the pull request:
- Unifies horizontal location of searchbox on home page and results page.
- Adds small space between search controls (sort, limit etc.) and range facet apply buttons.
- Improves horizontal alignment of sidebar items (particularly checkboxes when checkbox filters, OR-facets and multi-facet selection are enabled).
- Fixes and unifies searchbox's Search button and advanced search link in sandal5.
- Improves vertical alignment of icons, text and badges in active filters and facets.
- Makes search field reset button round.
- Makes account ajax alert background round in sandal5.
- Removes unused rules.

User content area:
- Stops dates from wrapping anywhere.
- Aligns checkboxes.
- Fixes checkboxes to display in narrow mode.
- Adds a bit of bottom margin to toolbar.

EreMaijala committed:
Improve alignment and spacing of various UI elements.

- Unifies horizontal location of searchbox on home page and results page.
- Adds small space between search controls (sort, limit etc.) and range facet apply buttons.
- Improves horizontal alignment of sidebar items (particularly checkboxes when checkbox filters, OR-facets and multi-facet selection are enabled).
- Fixes and unifies searchbox's Search button and advanced search link in sandal5.
- Improves vertical alignment of icons, text and badges in active filters and facets.
- Makes search field reset button round.
- Makes account ajax alert background round in sandal5.
- Removes unused rules.

EreMaijala commented:
@sturkel89 Would you be able to review this? The first comment should contain all that's changed, and of course nothing should look worse than before. :)

EreMaijala committed:
Tweak user content area.

- Stops dates from wrapping anywhere.
- Aligns checkboxes.
- Fixes checkboxes to display in narrow mode.
- Adds a bit of bottom margin to toolbar.

sturkel89 reviewed (commented):
I've compared test and dev using the Chrome browser on Windows. 

In some cases, I see the changes you've listed, but in other cases I can't detect the changes. Some of the differences may be browser-related.

-----------

### DEFINITELY FIXED:

First, the good news. I was able to confirm that the following bullet points have been fixed in the test branch.

- Makes search field reset button round.
- Adds small space between search controls (sort, limit etc.) and range facet apply buttons.
- Makes account ajax alert background round in sandal5.

------------

### USER AREA OBSERVATIONS (CHROME)

I was not able to detect any of these bullet point fixes listed for the My Account area.

- Aligns checkboxes.
- Fixes checkboxes to display in narrow mode.
- Adds a bit of bottom margin to toolbar.

I see that spacing of the user area sidebar is tighter vertically in the test branch. Spacing is also tighter vertically for all of the filters in the main VuFind window.

I don't see any problem with checkbox alignment in the user area in test or dev. I'm looking at the checkboxes for selecting retrieval requests to cancel, etc.; am I looking in the right place?

**- Stops dates from wrapping anywhere.**

In **Firefox**, I see that as long as the top menu is wide enough to be visible  (not scrunched into the hamburger menu), all dates in the Fines area do not wrap.  In this example, the browser is its narrowest width before jumping to hamburger.

<img width="800"  alt="image" src="https://github.com/user-attachments/assets/8830e79c-0e87-4a51-bbd0-2a08e273acb1" />

But in **Chrome**, the browser has to be pretty narrow for the top menu to go into hamburger mode. There are many widths where the top menu is visible but dates are wrapped in the Fines area; see this screenshot, with the round alert icon to show it's the test branch:

<img width="800"  alt="image" src="https://github.com/user-attachments/assets/c4f36f1d-23f9-4a30-8815-21da47d498fa" />

------------

### SEARCHBOX OBSERVATIONS ON CHROME

**-Unifies horizontal location of searchbox on home page and results page.**

I'm not sure I understand this one. In Sandal5, the search box on the home page is centered in a large background area, and in search results or an item record it is near the top of the page. 

In Sandal5, it appears to me that the search box is very stable between item record and search results view. It doesn't move horizontally or vertically. The same is true in test and dev.

In Bootstrap5, the search box is at the top of the home page, the results list page, and the item record page. I see the search box and the VuFind title moving back and forth horizontally when I toggle between the home page view and results list view. The home page is stable between search results view and item record view. The same is true in test and dev.

**Pixel in search box area:**

**Sandal5**
Spacing of searchbox elements look the same in test and dev. I see the same pixel between the searchbox and the dropdown in both cases.

<img width="654" height="366" alt="image" src="https://github.com/user-attachments/assets/ee6914da-c5c7-49fb-ae67-84b47a4c85aa" />

**Bootstrap5**
On the **home page**, there's no padding at all between the highlighted search box and the dropdown. This is true on test and dev.

<img width="631" height="166" alt="image" src="https://github.com/user-attachments/assets/1b1b3b3c-f3dd-4f66-91b3-269738987a39" />

On **search results and item record pages**, on **TEST**, there is still no space between the highlighted search box and the dropdown.

<img width="582" height="159" alt="image" src="https://github.com/user-attachments/assets/f667e0ff-37dc-4367-9d1d-69b36c92605b" />

On search results and item record pages, on **DEV**, there is a pixel of space between the highlighted search box and the dropdown. So, this is a change, but it's a very tiny pixel. Is this what's intended?

<img width="588" height="153" alt="image" src="https://github.com/user-attachments/assets/32febeee-d1d0-45d6-9d6e-810e5b3c8821" />

---------------

### OTHER OBSERVATIONS

I didn't see a difference in these areas, mostly focusing on testing in Sandal5. 

- Improves horizontal alignment of sidebar items (particularly checkboxes when checkbox filters, OR-facets and multi-facet selection are enabled).
- Improves vertical alignment of icons, text and badges in active filters and facets. 

I don't see this - the relative alignment of all of these things looks identical in test and dev to me, and they look fine. I checked both browsers; no notable differences due to browser.

The main difference between test and dev that I see has to do with vertical spacing in the sidebar. There's less vertical padding between items on filter lists in test. This is clear in both Chrome and Firefox.

These are Firefox screenshots:
<img width="648" height="593" alt="image" src="https://github.com/user-attachments/assets/80945806-9465-4cf4-8155-e145b0ccaec1" />

EreMaijala committed:
Merge remote-tracking branch 'origin/dev' into dev-ui-tweaks

EreMaijala committed:
Tweak checkboxes, toolbars and tables in MyResearch more.

EreMaijala committed:
Restore vertical padding of facets.

EreMaijala committed:
Use narrow layout also for online payment administration.

EreMaijala committed:
Fix focus ring color for select fields in sandal5.

EreMaijala commented:
@sturkel89 Thanks for testing and reporting!

My comments (I hope I didn't miss anything!):

- In MyResearch the alignment issues show up most prominently the new "Manage User Content" section, where the "Select All" checkbox was out of alignment with other buttons. They should be aligned like this:
<kbd><img width="159" height="450" alt="kuva" src="https://github.com/user-attachments/assets/c7925d5e-724b-45ad-8573-b91b0a55d9b5" /></kbd>

- I've now tweaked the fines table and toolbar to make it similar to others and to switch to the "narrow" layout for all but the widest screens. Otherwise there's just not enough space to display the content properly. I applied this also to online payment admin table (/Admin/Payment)
- "Unifies horizontal location of searchbox on home page and results page." is easiest to see on wide screens and boostrap5 theme. On the [advanced demo](https://vufind.org/advanced_demo/) home screen you can see the search controls with a pretty big left margin:
<kbd><img width="309" height="405" alt="kuva" src="https://github.com/user-attachments/assets/75e8f660-6a6e-4b3b-9269-2b91c5ed3ff8" /></kbd>
compared to results:
<kbd><img width="355" height="179" alt="kuva" src="https://github.com/user-attachments/assets/ec05b7a3-9561-43a7-a5c3-7e7ffc941637" /></kbd>
- In searchbox this fixes an apparent height difference of the Search button compared to the fields in sandal5 (not real, because it's actually caused by the dark background masking the border of the search fields, but it looks like the button is slightly higher):
<kbd><img width="267" height="68" alt="kuva" src="https://github.com/user-attachments/assets/24962023-963e-4766-a92c-10ad2e73b87f" /></kbd>
- Also in searchbox the hover color for Advanced was wrong in results.
- Also in searchbox colors are now the same regardless of whether you are on the home page or results etc.
- No padding between focused search terms field and dropdown is sort of expected. The controls are tightly coupled, and the focus ring needs the space. We could add more space between the controls, but then they'd be rather far apart when nothing's focused, and the Advanced link would need to be moved even further to keep it separate. Note: I've now fixed the focus ring color for the dropdowns in sandal5.
- Vertical space in facets now fixed.
- The horizontal alignment in facets can only be seen when checkbox facets and/or multi-facet selection is enabled. Should align these:
<kbd><img width="315" height="564" alt="kuva" src="https://github.com/user-attachments/assets/64d5cd5f-fa4a-44ea-a85c-faa822fff92b" /></kbd>
- The vertical alignment in facets can be seen also in your screenshot:
<kbd><img width="638" height="217" alt="kuva" src="https://github.com/user-attachments/assets/2cc5dd36-d569-4a6e-8796-da4dfa004f26" /></kbd>

sturkel89 reviewed (commented):
Thanks for the explanations and illustrations! I see all the changes and improvements you're talking about.

I just have two little concerns left.

-----

### FINES TABLE

 I see that the display switches to the "narrow" layout very quickly. It bothers me a little bit that the full top menu vs. hamburger menu are not tied to the switch between full screen and narrow display, but no user will care about that. :) 

The date columns look great in both themes on Firefox, and in Bootstrap5 on Chrome. In Sandal5 on Chrome, only the first date column consistently displays unwrapped (non-wrapped?) as in my screenshot above. However, I think the wrapped dates are not that big a deal. If it is very difficult to force them to look in that theme on Chrome as they look in the rest of the cases, we can merge as is.

-----

### ADVANCED SEARCH LINK

I see the change from unreadable dark blue against the background color in Sandal to the new white link in that theme. In Bootstrap5, the Advanced link doesn't change color at all on hover in the test branch. 

However, it seems that every other link marked by underlined text on the page changes color on hover, so it seems inconsistent that this one doesn't. Is there another way to have that link change appearance on hover (beyond the cursor turning to a finger) in either theme? 

I think that behavior should match that of the rest of the links in Bootstrap5, at least, since there is no hit to legibility in that theme if the link color changes on hover.

------

Sorry for the nitpicks! Overall, you've improved several very visible things!

EreMaijala committed:
Tweak btn-link colors, sandal5 background colors for contrast.

EreMaijala committed:
Tweak sandal5 advanced search link a bit more.

EreMaijala commented:
@sturkel89 Yes, the fines table switch happens early, but it's unavoidable with the amount of content displayed when trying to avoid crazy narrow columns. An option would be to never display the wide layout, but I don't really like that either.

I tweaked the Advanced search link to look more like normal links. For bootstrap5 it's now good, I think. sandal5 is a bit more difficult. I tried to give it a subtle color change on hover, but it looked muted if not white when not hovered over. In the end I decided to add a translucent background on hover. What do you think?

Regarding dates wrapping, I believe it's because Chrome treats the normal hyphen as breakable. You could configure VuFind to use non-breaking hyphen for display with this in config.ini, Site section:
`displayDateFormat = 'm‑d‑Y'
`
However, I haven't tested that enough to say it doesn't cause any odd effects anywhere. Another option would be to use a specific style for date columns that prevents word-breaks completely. Not going to happen here, because it should be applied consistently and would require too much effort. :)

EreMaijala commented:
@sturkel89 I made a couple of tweaks to background colors for improved contrast. Axe was complaining about too low contrast for the page header items and number badges in active filters, so I changed the background colors for these to a notch darker version. Now any remaining issues Axe reports are the same for bootstrap5 and sandal5.

sturkel89 reviewed (approved):
I didn't change anything on my end, but the dates all look good in the Fines table on Chrome now! The hyphens are non-breaking, and when you I narrow the screen from there it jumps directly to the narrow screen layout. There is no view where the dates are wrapped. (I checked my config.ini, and I do have displayDateFormat set as 'm-d-Y'.)

I like the translucent background on hover for the Advanced search link! That takes care of my complaint for Sandal5. In Bootstrap5, the Advanced search link now changes color on hover in the same way that all other links on the page change color. (That's a pretty subtle change, to be sure! What does Axe say about that?) I know that many libraries will customize Bootstrap5 and could have more extreme color changes associated with hover.

I see that the number badges in the filter area in Sandal5 are a bit darker now on test vs. dev. I am not sure I see the differences with the page header items, but I don't have any complaints! 

Things are looking better. If there isn't anything else on your list to check, I'm fine with merging this PR!
